idl/internal/thrift.y

```diff
@@ -358,18 +375,18 @@ throws
  ***************************************************************************/
 
 type
-    : lineno base_type_name type_annotations
-        { $$ = ast.BaseType{ID: $2, Annotations: $3, Line: $1} }
+    : pos base_type_name type_annotations
+        { $$ = ast.BaseType{ID: $2, Annotations: $3, Line: $1.Line, Column: $1.Column} }
 
     /* container types */
-    | lineno MAP '<' type ',' type '>' type_annotations
-        { $$ = ast.MapType{KeyType: $4, ValueType: $6, Annotations: $8, Line: $1} }
-    | lineno LIST '<' type '>' type_annotations
-        { $$ = ast.ListType{ValueType: $4, Annotations: $6, Line: $1} }
-    | lineno SET '<' type '>' type_annotations
-        { $$ = ast.SetType{ValueType: $4, Annotations: $6, Line: $1} }
-    | lineno IDENTIFIER
-        { $$ = ast.TypeReference{Name: $2, Line: $1} }
+    | pos MAP '<' type ',' type '>' type_annotations
+        { $$ = ast.MapType{KeyType: $4, ValueType: $6, Annotations: $8, Line: $1.Line, Column: $1.Column} }
+    | pos LIST '<' type '>' type_annotations
+        { $$ = ast.ListType{ValueType: $4, Annotations: $6, Line: $1.Line, Column: $1.Column} }
+    | pos SET '<' type '>' type_annotations
+        { $$ = ast.SetType{ValueType: $4, Annotations: $6, Line: $1.Line, Column: $1.Column} }
+    | lineno pos IDENTIFIER
```

Comment:
I could use some guidance here.

This is the last remaining occurrence of `lineno`. It's value isn't used by this expression, but when I remove it, I start getting incorrect column numbers in the resulting nodes. There's something "off" about the way the parser generates the code in that case, but I haven't been able to debug the root cause yet.

Any ideas would be welcome. I'm hopefully just overlooking something simple.

Reply:
I think there's now a parsing ambiguity between `type: pos base_type_name type_annotations` and `type: pos IDENTIFIER` when the `lineno` term is removed from the match.

Reply:
This is weird. It appears to be picking the position of the field rather than the position of the type.

Reply:
`pos IDENTIFIER` was always taking precedence over `pos base_type_name`. I was able to "fix" that by using `IDENTIFIER pos` in this one place, in the interest of practicality over purity.

I tried yacc's various precedence-controlling features without luck. I think something like [Selection Preferences](https://www.ibm.com/docs/en/zos/2.3.0?topic=topics-selection-preference-rules) would be ideal here, but goyacc doesn't support them.

```yacc
| ps IDENTIFIER [ ^ BOOL BYTE I8 I16 I32 I64 DOUBLE STRING BINARY ]
```

I couldn't think of another way to reorganize the grammar itself to avoid this situation, but I'm definitely open to ideas.

Reply:
I also tried to "trick" the grammar using an intermediate type, but I (helpfully) collapsed it, resulting in the same original problem:

```yacc
typeref
    : pos IDENTIFIER
        { $$ = ast.TypeReference{Name: $2, Line: $1.Line, Column: $1.Column} }
```